datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -1349,10 +1349,18 @@ private void handleLeaderDoAssignment(boolean cleanUpOrphanNodes) {
         failedStreams = unclaimedTokens.keySet();
         Set<String> hosts = unclaimedTokens.values().stream().flatMap(List::stream).
             map(AssignmentToken::getIssuedFor).collect(Collectors.toSet());
-        _log.error("Stop failed to propagate within {}ms for streams: {}. The following hosts failed to claim their token(s): {}",
-            _config.getStopPropagationTimeout(), failedStreams, hosts);
+
+        if (!isNewlyElectedLeader) {
+          _log.error("Stop failed to propagate within {}ms for streams: {}. The following hosts failed to claim their token(s): {}",
+              _config.getStopPropagationTimeout(), failedStreams, hosts);
+          _metrics.updateMeter(CoordinatorMetrics.Meter.NUM_FAILED_STOPS, failedStreams.size());
```

Comment:
In case of multiple stopping datastreams, even if one datastream fails to stop, all of them will be marked as failed stop here. Should we fix that?

Reply:
oops, I think I overlooked this. this is correct!